test/cudnn/nnlib.jl

```diff
@@ -1,139 +0,0 @@
-            @test testf((x, w) -> NNlib.conv(x, w, cdims; alpha=2.0), x, w)
-            @test testf((y, w) -> NNlib.∇conv_data(y, w, cdims; alpha=2.0), y, w)
-            @test testf((x, y) -> NNlib.∇conv_filter(x, y, cdims; alpha=2.0), x, y)
-            
-            @test testf((y, x, w) -> NNlib.conv!(copy(y), x, w, cdims; beta=2.0), y, x, w)
-            # @test testf((x, y, w) -> NNlib.∇conv_data!(copy(x), y, w, cdims; beta=2.0), x, y, w)
-            @test testf((w, x, y) -> NNlib.∇conv_filter!(copy(w), x, y, cdims; beta=2.0), w, x, y)
-
-            # Test the compatibility shims
-            cy,cx,cw = CuArray{Float32}.((y,x,w))
-            opts2 = Dict((k==:padding ? :pad : k)=>v for (k,v) in opts)
-            @test NNlib.conv!(similar(cy),cx,cw; opts2...) ≈ NNlib.conv!(similar(cy),cx,cw,cdims)
-            @test NNlib.∇conv_filter!(similar(cw),cy,cx; opts2...) ≈ NNlib.∇conv_filter!(similar(cw),cx,cy,cdims)
-        end
-
-        # Test that pooling is equivalent across GPU/CPU
-        pdims = PoolDims(x, 2)
-        y = maxpool(x, pdims)
-        dy = ones(size(y))
-        @test testf(x -> maxpool(x, pdims), x)
-        @test testf((dy, y, x) -> ∇maxpool(dy, y, x, pdims), dy, y, x)
-        @test testf(x -> maxpool(x, pdims), x)
-        @test testf((dy, y, x) -> ∇maxpool(dy, y, x, pdims), dy, y, x)
-
-        # Test the compatibility shims for pooling
-        cx,cy,cdy = CuArray{Float32}.((x,y,dy))
-        win,pad=2,1
-        @test maxpool!(similar(cy), cx, win; pad=pad, stride=win) ≈ maxpool!(similar(cy), cx, PoolDims(cx, win; padding=pad, stride=win))
-        @test meanpool!(similar(cy), cx, win; pad=pad, stride=win) ≈ meanpool!(similar(cy), cx, PoolDims(cx, win; padding=pad, stride=win))
-
-        # CPU implementation of ∇conv_bias!
-        db = zeros(Float64, 1, 1, 3, 1)
-        dy = randn(Float64, 8, 8, 3, 1)
-        function CUDNN.∇conv_bias!(db, dy)
-            db .= sum(dy, dims=(1:(ndims(dy)-2)))
-            return db
-        end
-        @test testf(CUDNN.∇conv_bias!, db, dy)
-    end
-
-    for dims in [(5,5), (5,)]
-        x = randn(Float64,dims)
-        y = softmax(x)
-        dy = randn(Float64,dims)
-        @test testf(softmax, x)
-        @test testf(∇softmax, dy, x) # add y when NNlib implements it
-        y = logsoftmax(x)
-        @test testf(logsoftmax, x)
-        @test testf(∇logsoftmax, dy, x) # add y when NNlib implements it
-    end
-end
-
-@testset "Activations and Other Ops" begin
-    @test testf(CUDNN.cudnnAddTensor, CUDA.rand(Float32, 10, 10, 3, 1), CUDA.rand(Float32, 10, 10, 3, 1))
-    @test testf(CUDNN.cudnnActivationForward!, CUDA.rand(Float32, 10, 10, 3, 1), CUDA.rand(Float32, 10, 10, 3, 1))
-    # @denizyuret: no high level api for backward functions, see CUDA/lib/cudnn/README.md
-    # @test testf(CUDNN.cudnnActivationBackward, CUDA.rand(Float32, 10, 10, 3, 1), CUDA.rand(Float32, 10, 10, 3, 1), CUDA.rand(Float32, 10, 10, 3, 1), CUDA.rand(Float32, 10, 10, 3, 1))
```

Comment:
@denizyuret should these stay in this repo?

Reply:
No need, these are tested in CUDA/test/cudnn/inplace.jl and activation.jl.